types/uuid/index.d.ts

```diff
@@ -8,10 +8,73 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.8
 
-import { v1, v3, v4, v5, validate } from './interfaces';
+// Uses ArrayLike to admit Unit8 and co.
+type OutputBuffer = ArrayLike<number>;
+type InputBuffer = ArrayLike<number>;
 
+interface RandomOptions {
```

Comment:
You might want to export these interfaces so that they are reachable, i.e.:

```ts
import { v1, V1Options } from "uuid";

const options: V1Options = {
  clockseq: 0,
  msecs: Date.now()
};

v1(options);
```

Otherwise anyone using the types would have to repeat the definitions themselves since they're not exported.

Reply:
In this PR I was focusing on adding the new APIs that the upstream `uuid` module added in the latest version. The options objects have so far not been exposed, so while I agree that this is probably nice to have, it's at least not a regression with respect to the current state of the types.

Could we maybe first get this merged since people are waiting for it https://github.com/uuidjs/uuid/issues/502 and keep submitting incomplete PRs for these missing APIs https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46477 & https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46557?

I'm of course happy to submit the proposed improvements in another PR, but I'd really prefer to get the types in sync with the upstream module API as quickly as possible.

Reply:
@rbuckton I have exported the V1Options and V4Options now. Do you think there's any value in exposing the sub-types that are used to compose these two types as well?